qemu/tests/qemu_guest_agent.py

```diff
@@ -752,44 +753,140 @@ def get_allocation_bitmap():
+
+        Steps:
+        1) login guest with serial session
+        2) get the available interface name via mac address
+        3) check the available interface name is the same with guest
+        4) check ip address is the same with guest
+        5) create a bridge interface for linux guest and check it from guest agent;
+           disable interface for windows guest and check it from guest agent
+        6) check "guest-network-get-interfaces" result
+        7) recover the interfaces
+        8) change ip address
+        9) check "guest-network-get-interfaces" result
+
         :param test: kvm test object
         :param params: Dictionary with the test parameters
         :param env: Dictionary with test environment.
         """
-        def find_interface_by_name(interface_list, target_interface):
+        def get_interface(ret_list, mac_addr):
             """
-            find the specific network interface in the interface list return
-            by guest agent. return True if find successfully
+            Get the available interface name.
+
+            :return: interface name and the interface's index in ret_list
             """
-            for interface in interface_list:
-                if target_interface == interface["name"]:
-                    return True
-            return False
-        session = self._get_session(params, None)
+            interface_name = ""
+            index_if = 0
+            for interface in ret_list:
+                if "hardware-address" in interface and \
+                                interface["hardware-address"] == mac_addr:
+                    interface_name = interface["name"]
+                    break
+                index_if += 1
+            return interface_name, index_if
 
-        # check if the cmd "guest-network-get-interfaces" work
+        def ip_addr_check(session, mac_addr, ret_list, index_if, interface_name):
+            """
+            Check the ip address from qga and guest inside.
+
+            :param session: serial session
+            :param mac_addr: mac address of nic
+            :param ret_list: return list from qg
+            :param index_if: the interface's index in ret
+            :param interface_name: interface name
+            :return:
+            """
+            os_type = self.params.get("os_type")
+            guest_ip_addr_ipv4 = utils_net.get_guest_ip_addr(session, mac_addr, os_type)
+            guest_ip_addr_ipv6 = utils_net.get_guest_ip_addr(session, mac_addr, os_type,
+                                                             ip_version="ipv6",
+                                                             linklocal=True)
+            ip_lists = ret_list[index_if]["ip-addresses"]
+            index_ip = 0
+            for ip in ip_lists:
+                if ip["ip-address-type"] == "ipv4":
+                    ip_addr_qga_ipv4 = ip_lists[index_ip]["ip-address"]
+                if ip["ip-address-type"] == "ipv6":
```

Comment:
Could here be `else`? It will check condition once instead of twice for two `if`.